app/src/main/java/com/example/travelapp/DetailsAdapter.java

```diff
@@ -0,0 +1,67 @@
+package com.example.travelapp;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+
+public class DetailsAdapter extends RecyclerView.Adapter<DetailsAdapter.ViewHolder> {
+
+    private static final String TAG = "DetailsAdapter";
+    private Context context;
+    private List<String> destinations;
+    private List<String> distances;
+
+    public DetailsAdapter(Context context, Details details) {
```

Comment:
why not just make `Details` an interface? that way you can extend` RecyclerView` and implement `Details` without having to bring Details in with the constructor?